pkg/scheduler/schedule_one_test.go

```diff
@@ -953,6 +980,17 @@ func TestSchedulerScheduleOne(t *testing.T) {
 						t.Errorf("Expected unschedulable pods to be empty, but it's not.\nGot: %v", unschedulablePods)
 					}
 				}
+				if item.expectError != nil {
+					expectedNominatingInfo := item.expectNominatingInfo
+					// Check nominatingInfo expectation based on feature gate
+					if !nominatedNodeNameForExpectationEnabled && expectedNominatingInfo == nil {
+						expectedNominatingInfo = &framework.NominatingInfo{NominatingMode: framework.ModeOverride, NominatedNodeName: ""}
+					}
+					if !reflect.DeepEqual(expectedNominatingInfo, gotNominatingInfo) {
+						t.Errorf("Expected nominatingInfo to be %v, but got %v (NominatedNodeNameForExpectation enabled: %v)",
+							expectedNominatingInfo, gotNominatingInfo, nominatedNodeNameForExpectationEnabled)
```

Comment:
Feature gate state is already added to the test case name
```suggestion
						t.Errorf("Expected nominatingInfo to be %v, but got %v", expectedNominatingInfo, gotNominatingInfo)
```